### ISSUE TITLE
[UI] Hooks view empty state

### ISSUE THREAD

**Issue by helfi92** [taskcluster/taskcluster#2606]
**To Reproduce**

1. Go to https://firefox-ci-tc.services.mozilla.com/hooks
2. Search for something that doesn't exist (e.g., foobar)

**Expected behavior**
Some text telling the user that there are no hooks to be shown with this search term.

**Actual**
An empty screen.

<img width="1490" alt="Screen Shot 2020-04-06 at 8 31 26 AM" src="https://user-images.githubusercontent.com/3766511/78558670-04cb4a00-77e1-11ea-982f-31e8cf15bf03.png">

**Taskcluster version**
28.0.0

**Comment by ashishzingh** [comment]
i would like to work on this issue

**Comment by helfi92** [comment]
Go for it.

**Comment by helfi92** [comment]
How is it going? :-)

**Comment by ashishzingh** [comment]
i am trying to figure out how to access the subtree in material ui once the search is performed.

**Comment by aryamanpuri** [comment]
Yes, we can show like this :
![Screenshot from 2020-04-10 12-45-33](https://user-images.githubusercontent.com/43513114/78971454-3c136100-7b29-11ea-845e-0fa5fefda222.png)

**Comment by ashishzingh** [comment]
@aryamanpuri  do we need to access the subtree to find out whether the search has no results or is there any other method?

**Comment by ashishzingh** [comment]
@helfi92 can you please guide me how to find whether the subtree on searching inside <MuiTreeView> is empty or not.
https://github.com/taskcluster/taskcluster/blob/796b4ed7273c005cf58bfac461ab004bf30f6225/ui/src/views/Hooks/ListHooks/index.jsx#L90-L100
hookGroups remain constant even on searching.

**Comment by helfi92** [comment]
This one is a bit not like the other. If we want to render something when the list is empty, we have two choices:

1. Use `forwardRef` as defined in https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-class-component.

2. Expose a new prop on material-ui-treeview (e.g., `onEmptySearch`) that would accept a react element to render when the tree is not empty but the search term finds no result.

I think the second strategy is a bit easier and feels less hacky. Here's what needs to be done:

1. Add a prop to the component in https://github.com/helfi92/material-ui-treeview/ that would render when we have a tree but the search term makes it have nothing to render.
2. Create a PR and wait for me to review and merge it. Once merged, I will publish a new version of that library.
3. Update the version of `material-ui-treeview` here and make the necessary changes:

```jsx
<MuiTreeView ... onEmptySearch={<Typography variant="subtitle1">No items for search term {hookSearch}</Typography>} />
```

Hope that helps.

**Comment by helfi92** [comment]
Should be fixed in material-ui-treeview@4.2.1 thanks to @ashishzingh 💯
